packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/redshift-cluster.ts

```diff
@@ -0,0 +1,276 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as firehose from '@aws-cdk/aws-kinesisfirehose';
+import * as kms from '@aws-cdk/aws-kms';
+import * as redshift from '@aws-cdk/aws-redshift';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { BackupMode, CommonDestinationProps, CommonDestinationS3Props, Compression } from './common';
+import { createBackupConfig, createBufferingHints, createEncryptionConfig, createLoggingOptions, createProcessingConfig } from './private/helpers';
+import { FirehoseRedshiftTable } from './redshift/table';
+import { FirehoseRedshiftUser } from './redshift/user';
+
+/**
+ * The Redshift user Firehose will assume to deliver data to Redshift
+ */
+export interface RedshiftUser {
+  /**
+   * Username for user that has permission to insert records into a Redshift table.
+   */
+  readonly username: string;
+
+  /**
+   * Password for user that has permission to insert records into a Redshift table.
+   *
+   * Do not put passwords in your CDK code directly.
+   *
+   * @default - a Secrets Manager generated password.
+   */
+  readonly password?: cdk.SecretValue;
+
+  /**
+   * KMS key to encrypt the generated secret.
+   *
+   * @default - default master key.
+   */
+  readonly encryptionKey?: kms.IKey;
+}
+
+/**
+ * A column in a Redshift table.
+ */
+export interface RedshiftColumn {
+  /**
+   * The name of the column.
+   */
+  readonly name: string;
+
+  /**
+   * The data type of the column.
+   */
+  readonly dataType: string;
+}
+
+/**
+ * Properties for configuring a Redshift delivery stream destination.
+ */
+export interface RedshiftDestinationProps extends CommonDestinationProps, CommonDestinationS3Props {
+  /**
+   * The cluster user that has INSERT permissions to the desired output table.
+   */
+  readonly user: RedshiftUser;
+
+  /**
+   * The database containing the desired output table.
+   */
+  readonly database: string;
+
+  /**
+   * The table that data should be inserted into.
+   */
+  readonly tableName: string;
+
+  /**
+   * The table columns that the source fields will be loaded into.
+   */
+  readonly tableColumns: RedshiftColumn[];
+
+  /**
+   * The secret that holds Redshift cluster credentials for a user with administrator privileges.
+   *
+   * Used to create the user that Firehose assumes and the table that data is inserted into.
+   *
+   * @default - the master secret is taken from the cluster.
+   */
+  readonly masterSecret?: secretsmanager.ISecret;
```

Comment:
If this is specified, why would the user also need to specify a 'RedshiftUser'? If they specify a secret that is the "Credentials for a Redshift cluster" type, then would the username and password and encryption key all be there as part of the secret? 

Can we enforce that the secret passed in here follows the "Credentials for a Redshift cluster" format? This is defined in the AWS console, but does not seem to be defined in CFN.

We should choose an inclusive name for this term: `adminSecret`, `superUserCredentials`?